[GeomechanicsApplication] Add documentation for compression cap (#14348)
KratosMultiphysics/Kratos: applications/GeoMechanicsApplication/custom_constitutive/README_cap.md

Reviewer: This text is difficult to understand for someone with no experience in constitutive models. I think this should be broken down into the fact that we want to be able to introduce a cap. After the introduction of the cap we should discuss the option of hardening/softening of the cap which is a follow-up development. Is it a cap or a compression cap?

Reviewer: ```suggestion
The standard Mohr–Coulomb yield surface characterizes shear failure in geomaterials by relating the shear stress $\tau$ on a potential failure plane to the corresponding normal stress $\sigma$. While suitable for frictional materials, the standard Mohr-Coulomb envelope lacks a mechanism to limit the admissible stress space under high compressive pressure. Most materials do have a limited admissible stress under high compressive pressure. Therefor it is necessary to be able to cap the compressive pressure in constitutive models. The standard Mohr-Coulomb model does not have this ability and cannot represent a compaction limit, the onset for crushing, or hardening behavior that occurs in soils and rocks under high confining stresses. 

In Figure .. we have shown the contours of the cap within the Mohr-Coulomb model. This cap is a radial yield surface that joins the shear yield surface and results in a smooth closure of the yield surface in the high-compression regime. Here we describe the combined Mohr-Coulomb and cap yield surfaces.
```

Reviewer: ```suggestion
- $`\sigma`$ = normal stress
```

Reviewer: ```suggestion
- $`\tau`$ = shear stress 
```

Reviewer: ```suggestion
- $`c`$ = cohesion
```

Reviewer: What is the added value of this sentence? I think it would be good that we describe the difference between (sigma,tau), (p,q), (sigma1,sigma2,sigma2) stress spaces somewhere. It would be helpful to visualize the stress spaces for each configuration.

Reviewer: ```suggestion
### Cap concept in constitutive models
```

Reviewer: entire text of Compression cap concept can be removed

Reviewer: ```suggestion
A cap yield surface introduces a limit to admissible volumetric compression and establishes a mechanism for volumetric plastic deformation. The shape of a cap yield surface depends on the stress space that is used. In this case the $'p-q'$ stress-invariant stress-space is used where the cap is defined as an ellipse of which the Mohr-Coulomb yield surface encloses the compressive regime (limit) (Reason for using p,q should be stated here). In mathematical form, an elliptical cap can be defined as:
```

Reviewer: Does X define the cap size or the elliptic shape of the cap?

Reviewer: The description of hardening for the cap is confusing at this stage. I would a new section of the hardening of a cap with an updated figure in which the hardening behaviour is visualized. Preferably in a separate document (readme_hardening) or at the end of this document.

> In stress-invariant form, the MC yield function is typically written as:

Reviewer: ```suggestion
In stress-invariant form (p,q) stress space, the MC yield function is typically written as:
```

```math

>     F_{MC}(p, q) = q + \frac{6 \sin{\phi}}{3 - \sin{\phi}} p - \frac{6 c \cos⁡{\phi}}{3 - \sin{\phi}}

Reviewer 2: Note that when we take up Hans his advice , these formula's will include Lode angle theta. ( not for now. )

Here, we need to convert the compression cap yield surface from $(p, q)$ coordinates to $(\sigma, \tau)$ coordinates. The conversion is given in Appendix A.

### Plastic potential for the compression cap

For the cap branch, plastic deformation is primarily volumetric (compaction), and the plastic potential is usually taken to be associated. The flow function is then:

```math
    G_{cap} \left(p, q \right) = F_{cap} \left(p, q \right)
```
The derivative of the flow function is:

```math
    \frac{\partial G_{cap}}{\partial \sigma} = \frac{2 q}{X^2} \frac{\partial q}{\partial \sigma} + 2 p \frac{\partial p}{\partial \sigma}
```

It gives:

```math
    \frac{\partial G_{cap}}{\partial \sigma_i} = \frac{1}{3} \frac{\partial G_{cap}}{\partial p} + \frac{3}{2 q} \frac{\partial G_{cap}}{\partial q} \left(\sigma_i - p \right)
```

### Cap corner point

The point where the compression cap yield surface intersects the Mohr-Coulomb yield surface is called the cap corner point. This point can be calculated by solving the Coulomb yield surface for $q$:

```math
    q = - \frac{6 \sin{\phi}}{3 - \sin{\phi}} p + \frac{6 c \cos⁡{\phi}}{3 - \sin{\phi}}
```

> then subsituting in the cap yield surface, it leads to the following equation:

Reviewer: ```suggestion
then subsituting in the cap yield surface, it leads to the following quadratic equation:
```

```math

>     A p_{corner}^2 + B p_{corner} + C = 0

Reviewer: ```suggestion
    a p_{corner}^2 + b p_{corner} + c = 0
```

```

> 

Reviewer: ```suggestion
Through the abc-formula the roots of this quadratic equation can be solved for.
```

> Where,

Reviewer: ```suggestion
Where, a,b, and c can be replaced by the following terms A,B, and C, respectively.
```

> Here, we need to convert the compression cap yield surface from $(p, q)$ coordinates to $(\sigma, \tau)$ coordinates. The conversion is to be followed ...

Reviewer: How the conversion takes place should still be written

### Plastic potential for the compression cap

For the cap branch, plastic deformation is primarily volumetric (compaction), and the plastic potential is usually taken to be associated. The flow function is then:

```math
    G_{cap} \left(p, q \right) = F_{cap} \left(p, q \right)
```
The derivative of the flow function is:

```math
    \frac{\partial G_{cap}}{\partial \sigma} = \frac{2 q}{X^2} \frac{\partial q}{\partial \sigma} + 2 p \frac{\partial p}{\partial \sigma}
```

> or

Reviewer: Why is this 'or', this is not very clear to me, please elaborate

```math
    \frac{\partial G_{cap}}{\partial \sigma_i} = \frac{1}{3} \frac{\partial G_{cap}}{\partial p} + \frac{3}{2 q} \frac{\partial G_{cap}}{\partial q} \left(\sigma_i - p \right)
```

### Cap corner point

The point where the compression cap yield surface intersects the Mohr-Coulomb yield surface is called the cap corner point. This point can be calculated by solving the Coulomb yield surface for $q$:

```math
    q = - \frac{6 \sin{\phi}}{3 - \sin{\phi}} p + \frac{6 c \cos⁡{\phi}}{3 - \sin{\phi}}
```

then subsituting in the cap yield surface, it leads to the following equation:

```math
    A p_{corner}^2 + B p_{corner} + C = 0
```

Where,

```math
    A = 1 + \frac{1}{X^2} \left( \frac{6  \sin{\phi}}{3 - \sin{\phi}} \right) ^2
```
```math
    B = -\frac{72}{X^2} \frac{c \sin{\phi} \cos{\phi}}{\left( 3 - \sin{\phi} \right)^2}
```
```math
    C = \frac{1}{X^2} \left( \frac{6 c \cos{\phi}}{3 - \sin{\phi}} \right)^2 - p_c^2
```

A second order polynomial equation needs to be solved, and the minimum root needs to be selected,

```math
    p_{corner} = \frac{ -B - \sqrt{B^2 - 4 A C}}{2A}
```
then
```math

>     q_{corner} = -a_2 p + c_2

Reviewer 2: The definition of a_2 and c_2 was removed to have a formulation of A, B and C without intermediate variables. Please do that here too.

Author: Done

```

### Return mapping from cap compression zone
The cap compression zone is the region where the trial principal stresses are,
1. Outside the compression cap yield surface
```math
    \frac{q^2}{X^2} + p^2 - p_c^2 > 0
```
2. Under the line which passes from the cap corner point and in the direction normal to the flow function of the cap yield surface. 
```math
    q - q_{corner} - \left( G_{cap,p}/G_{cap,q} \right) (p - p_{corner}) < 0
```

Then the trial principal stresses need to be mapped to the cap yield surface by:
```math

>     \sigma = \sigma^{trial} + \lambda C \frac{\partial G_{cap}}{\partial \sigma}

Reviewer: Not sure I like the use of C here, maybe this should be mentioned in the text.

```

### Return mapping from cap corner zone

> The cap compression zone is the region where the trial $\left(p, q \right)$ stress invariants are demarcated by the following two lines,

Reviewer: is demarcated a word?

1. above the line which passes from the cap corner point and in the direction normal to the cap flow function $G_{cap}$. 
```math
    q - q_{corner} - \left( G_{cap,p}/G_{cap,q} \right) (p - p_{corner}) > 0
```

2. Under the line which passes from the cap corner point and in the direction normal to the Coulomb flow function $G_{MC}$. 
```math
    q - q_{corner} - \left( G_{MC,p}/G_{MC,q} \right) (p - p_{corner}) < 0
```

Then the trial principal stresses need to be mapped to the cap yield surface and Coulomb yield surface by:
```math
    \sigma = \sigma^{trial} + \lambda_{MC} C \frac{\partial G_{MC}}{\partial \sigma}
    + \lambda_{cap} C \frac{\partial G_{cap}}{\partial \sigma}
```

> Substuting this trial stresses in compression cap and Coulomb yield surfaces, it leads to two equations and two unknowns.

Reviewer: ```suggestion
Substituting these trial stresses in the compression cap and Coulomb yield surfaces, leads to two equations and two unknowns.
```

```math
    c_1 \lambda_{MC} + c_2 \lambda_{cap} = c_3
```
```math
    c_4 \lambda_{MC}^2 + c_5 \lambda_{cap}^2 + c_6 \lambda_{MC} + c_7 \lambda_{cap}
    + c_8 \lambda_{MC} \lambda_{cap} = c_9
```

Combining those two equations, it leads to a second order polynomial equation for $\lambda_{cap}$.

```math
    A \lambda_{cap}^2 + B \lambda_{cap} + C = 0
```

where,

```math
    c_0 = \frac{6 \sin{\phi}}{3 - \sin{\phi}}
```
```math
    \left[ p_{MC}^{cor}, q_{MC}^{cor} \right]^T = C G_{MC}
```
```math
    \left[ p_{cap}^{cor}, q_{cap}^{cor} \right]^T = C G_{cap}
```
```math
    c_1 = q_{MC}^{cor} + c_0 p_{MC}^{cor}
```
```math
    c_2 = q_{cap}^{cor} + c_0 p_{cap}^{cor}
```
```math
    c_3 = -F_{MC} \left( \sigma^{trial} \right)
```
```math
    c_4 = q_{MC}^{cor^2} / X^2 + p_{MC}^{cor^2}
```
```math
    c_5 = q_{cap}^{cor^2} / X^2 + p_{cap}^{cor^2}
```
```math
    c_6 = 2 \left( q^{trial} q_{MC}^{cor} / X^2 + p^{trial} p_{MC}^{cor} \right)
```
```math
    c_7 = 2 \left( q^{trial} q_{cap}^{cor} / X^2 + p^{trial} p_{cap}^{cor} \right)
```
```math

>     c_8 = 2 \left( q_{MC}^{cor} q_{cap}^{cor} / X^2 + p_{MC}^{cor} p_{cap}^{cor} \right)

Reviewer: This one appears to have a representation error in it

Author: I did double check. Please let me know which part of this equation, do you think it has error?

```
```math
    c_9 = -F_{cap} \left( \sigma^{trial} \right)
```

```math

>     A = \frac{c_2}{c_1} \left( \frac{c_2 c_4}{c_1} - c_8 \right) + c_5

Reviewer: You cannot use A, B and C again here. Would change this to D E and F

Reviewer 2: I strongly disagree with this. The ABC formula is used worldwide to find the roots of a quadratic equation. Renaming its characteristic parameters to anything else that ABC will not contribute to clarity. The sectioning of the file makes clear what is solved for.

Author: let's discuss it with @WPK4FEM 
The following points are all related to this naming convention too

Reviewer: Let's avoid the use of capitol letters in general and write down full equations. By stating that the said equation is a quadratic equation for which the root can be found through the abc-formula should suffice. This only requires that the quadratic form (ax2+bx+c) and the solution x = ... are written down at the point where the abc-formula is mentioned.

>     A = \frac{c_2}{c_1} \left( \frac{c_2 c_4}{c_1} - c_8 \right) + c_5

Reviewer: ```suggestion
    D = \frac{c_2}{c_1} \left( \frac{c_2 c_4}{c_1} - c_8 \right) + c_5
```

```
```math

>     B = \frac{1}{c_1} \left( -2 \frac{c_2 c_3 c_4}{c_1} - c_2 c_6 + c_3 c_8 \right) + c_7

Reviewer: ```suggestion
    E = \frac{1}{c_1} \left( -2 \frac{c_2 c_3 c_4}{c_1} - c_2 c_6 + c_3 c_8 \right) + c_7
```

```
```math

>     C = \frac{c_3}{c_1} \left( \frac{c_3 c_4}{c_1} + c_6 \right) - c_9

Reviewer: ```suggestion
    F = \frac{c_3}{c_1} \left( \frac{c_3 c_4}{c_1} + c_6 \right) - c_9
```

```
    
Then, solving the second order polynomial, it gives 
```math

>     \lambda_{cap} = \frac{-B + \sqrt{B^2 - 4 A C}}{2A}

Reviewer: ```suggestion
    \lambda_{cap} = \frac{-E + \sqrt{E^2 - 4 D F}}{2D}
```

```
```math
    \lambda_{MC} = \frac{c_3 - c_2  \lambda_{cap}}{c_1}
```

### Plastic multiplier for compression cap
The plastic multiplier above, $\lambda_{cap}$ needs to be calculated. The mapping is,
```math
    \sigma = \sigma^{trial} + \lambda_{cap} C \frac{\partial G_{cap}}{\partial \sigma}
```

Now, by considering the cap yield surface function, and extracting $p$ and $q$ in the form of $\sigma$'s,
```math
    \frac{1}{2 X^2} \left[ \left( \sigma_1 - \sigma_2 \right)^2 + \left( \sigma_2 - \sigma_3 \right)^2 + \left( \sigma_3 - \sigma_1 \right)^2 \right] + \frac{1}{9} \left( \sigma_1 + \sigma_2 + \sigma_3 \right)^2 - p_c^2 = 0
```
We define the following vectors,
```math
    \sigma_{diff} = \left[\sigma_1 - \sigma_2 \; , \; \sigma_2 - \sigma_3 \; , \; \sigma_3 - \sigma_1 \right]^T
```
```math
    \sigma_{diff}^{cor} = \left[\sigma_1^{cor} - \sigma_2^{cor} \; , \; \sigma_2^{cor} - \sigma_3^{cor} \; , \; \sigma_3^{cor} - \sigma_1^{cor} \right]^T
```

where
```math
    \sigma^{cor} = C \frac{\partial G_{cap}}{\partial \sigma}
```

Subsituting the stresses with the mapped stresses, we get the following relations.

> ```math

Reviewer: Again, this cannot be A B and C, as to follow this should be G, H, and I (KLM or M, N, P is also a suggestion)

>     A = \frac{\sigma_{diff}^{cor} \cdot \sigma_{diff}^{cor}}{2 X^2} + \frac{1}{9} \left( \sum_{i=1}^3{\sigma_i^{cor}} \right)^2

Reviewer: ```suggestion
    K = \frac{\sigma_{diff}^{cor} \cdot \sigma_{diff}^{cor}}{2 X^2} + \frac{1}{9} \left( \sum_{i=1}^3{\sigma_i^{cor}} \right)^2
```

```
```math

>     B = \frac{\sigma_{diff} \cdot \sigma_{diff}^{cor}}{X^2} + \frac{2}{9} \sum_{i=1}^3{\sigma_i} \sum_{i=1}^3{\sigma_i^{cor}}

Reviewer: ```suggestion
    L = \frac{\sigma_{diff} \cdot \sigma_{diff}^{cor}}{X^2} + \frac{2}{9} \sum_{i=1}^3{\sigma_i} \sum_{i=1}^3{\sigma_i^{cor}}
```

```
```math

>     C = F_{cap} \left( \sigma^{trial} \right)

Reviewer: ```suggestion
    M = F_{cap} \left( \sigma^{trial} \right)
```

```

the solution of this equation, gives:

 ```math

>     \lambda_{cap} = \frac{-B + \sqrt{B^2 - 4 A C}}{2A}

Reviewer: ```suggestion
    \lambda_{cap} = \frac{-L + \sqrt{L^2 - 4 K M}}{2K}
```

```math
    \frac{\partial G_{cap}}{\partial \sigma} = \frac{2 q}{X^2} \frac{\partial q}{\partial \sigma} + 2 p \frac{\partial p}{\partial \sigma}
```

or

```math
    \frac{\partial G_{cap}}{\partial \sigma_i} = \frac{1}{3} G_{,p} + \frac{3 G_{,q}}{2 q} \left(\sigma_i - p \right)

> ```

Reviewer 2: Here the derivative is done with , notation for the first time. That comes without explanation. I would prefer to still use \partial. That is what is on the left side too.

Author: @WPK4FEM 
Thank you for the review. 
They are changed to the form of partial.

### Cap corner point

The point where the compression cap yield surface intersects the Mohr-Coulomb yield surface is called the cap corner point. This point can be calculated by extracting $q$ of Coulomb yield surface:

```math
    q = - \frac{6 \sin{\phi}}{3 - \sin{\phi}} p + \frac{6 c \cos⁡{\phi}}{3 - \sin{\phi}}
```

then subsituting in the cap yield surface, it leads to the following equation:

```math
    A p_{corner}^2 + B p_{corner} + C = 0
```

Where,

```math
    A = 1 + b_1 a_2^2
```
```math
    B = -2 b_1 a_2 c_2
```
```math
    C = b_1 c_2^2 - c_1
```

```math
    b_1 = 1 / X^2
```
```math
    c_1 = p_c^2
```
```math
    a_2 = \frac{6  \sin{\phi}}{3 - \sin{\phi}}
```
```math
    c_2 = \frac{6 c \cos{\phi}}{3 - \sin{\phi}}

> ```

Reviewer 2: The many intermediate variable definitions are more a programming aid than making it more readable for me.

Author: I tried to show how the second order polynomian equation was derived, to be able to follow the process if it will be necessary in the future. So, I tried to include the intermediate process. What do you think about it?

Reviewer: I would start with the full equation instead of the ABC form. After the full statement you can mention that this takes the ABC form which can be solved.

Reviewer: I have made some additions, I disagree with the use of A, B, and C multiple times, it is confusing for readers.

> A second order polynomial equation needs to be solved, and the minimum root needs to be selected, because $p \lt 0$

Reviewer 2: The "because p < 0 " I don't understand. With large cohesion, I can create a situation where the cap-MC corner has positive p. Its more a takeaway from drawing the intersections that you want the one with lowest p.

Author: This sentence is removed.

```math
    p_{corner} = \frac{ -B - \sqrt{B^2 - 4 A C}}{2A}
```
then
```math
    q_{corner} = -a_2 p + c_2
```

### Return mapping from cap compression zone
The cap compression zone is the rigion where the trial principal stresses are,
1. Outside the compression cap yield surface
```math
    \frac{q^2}{X^2} + p^2 - p_c^2 > 0
```
2. Under the line which passes from the cap corner point and in the direction normal to the flow function of the cap yield surface. 
```math
    q - q_{corner} - \left( G_{cap,p}/G_{cap,q} \right) (p - p_{corner}) < 0
```

Then the trial principal stresses need to be mapped to the cap yield surface by:
```math
    \sigma = \sigma^{trial} + \lambda C \frac{\partial G_{cap}}{\partial \sigma}
```

### Return mapping from cap corner zone

> The cap compression zone is the rigion where the trial principal stresses are,

Reviewer 2: ```suggestion
The cap compression zone is the region where the trial (p, q) stress invariants are demarcated by the following two lines,
```

Author: Done

> 1. above the line which passes from the cap corner point and in the direction normal to the flow function of the cap yield surface. 

Reviewer 2: ```suggestion
1. above the line which passes from the cap corner point and in the direction normal to the cap flow function $G_cap$. 
```

Author: Done

```math
    q - q_{corner} - \left( G_{cap,p}/G_{cap,q} \right) (p - p_{corner}) > 0
```

> 2. Under the line which passes from the cap corner point and in the direction normal to the flow function of the Coulomb yield surface. 

Reviewer 2: ```suggestion
2. Under the line which passes from the cap corner point and in the direction normal to the Coulomb flow function $G_MC$ 
```

Author: Done

```math
    q - q_{corner} - \left( G_{MC,p}/G_{MC,q} \right) (p - p_{corner}) < 0
```

Then the trial principal stresses need to be mapped to the cap yield surface by:
```math
    \sigma = \sigma^{trial} + \lambda_{MC} C \frac{\partial G_{MC}}{\partial \sigma}
    + \lambda_{cap} C \frac{\partial G_{cap}}{\partial \sigma}
```

> Subsituting this traisl stresses in compression cap and Coulomb yield surfaces, it leads to two equations and two unknowns.

Reviewer 2: ```suggestion
Subsituting this trial stresses in compression cap and Coulomb yield surfaces, it leads to two equations and two unknowns.
```

Author: Done

```math
    A \lambda_{cap}^2 + B \lambda_{cap} + C = 0
```

where,

```math
    c_0 = \frac{6 \sin{\phi}}{3 - \sin{\phi}}
```
```math
    \left[ p_{MC}^{cor}, q_{MC}^{cor} \right]^T = C G_{MC}
```
```math
    \left[ p_{cap}^{cor}, q_{cap}^{cor} \right]^T = C G_{cap}
```
```math
    c_1 = q_{MC}^{cor} + c_0 p_{MC}^{cor}
```
```math
    c_2 = q_{cap}^{cor} + c_0 p_{cap}^{cor}
```
```math
    c_3 = -F_{MC} \left( \sigma^{trial} \right)
```
```math
    c_4 = q_{MC}^{cor^2} / X^2 + p_{MC}^{cor^2}
```
```math
    c_5 = q_{cap}^{cor^2} / X^2 + p_{cap}^{cor^2}
```
```math
    c_6 = 2 \left( q^{trial} p_{MC}^{cor} / X^2 + p^{trial} p_{MC}^{cor} \right)
```
```math
    c_7 = 2 \left( q^{trial} q_{cap}^{cor} / X^2 + p^{trial} p_{cap}^{cor} \right)
```
```math
    c_8 = 2 \left( q_{MC}^{cor} q_{cap}^{cor} / X^2 + p_{MC}^{cor} p_{cap}^{cor} \right)
```
```math
    c_9 = -F_{cap} \left( \sigma^{trial} \right)
```

```math
    A = \frac{c_2}{c_1} \left( \frac{c_2 c_4}{c_1} - c_8 \right) + c_5
```
```math
    B = \frac{1}{c_1} \left( -2 \frac{c_2 c_3 c_4}{c_1} - c_2 c_6 + c_3 c_8 \right) + c_7
```
```math
    C = \frac{c_3}{c_1} \left( \frac{c_3 c_4}{c_1} + c_6 \right) - c9
```
    

> Then, solving the recond order polynomial, it gives 

Reviewer 2: ```suggestion
Then, solving the second order polynomial, it gives 
```

Author: Done

```math
    \lambda_{cap} = \frac{-B + \sqrt{B^2 - 4 A C}}{2A}
```
```math
    \lambda_{MC} = \frac{c_3 - c_2  \lambda_{cap}}{c_1}
```

### Plastic multiplier for compression cap
The plastic multiplier above, $\lambda_{cap}$ needs to be calculated. The mapping is,
```math
    \sigma = \sigma^{trial} + \lambda_{cap} C \frac{\partial G_{cap}}{\partial \sigma}
```

Now, by considering the cap yield surface function, and extracting $p$ and $q$ in the form of $\sigma$'s,
```math
    \frac{1}{2 X^2} \left[ \left( \sigma_1 - \sigma_2 \right)^2 + \left( \sigma_2 - \sigma_3 \right)^2 + \left( \sigma_3 - \sigma_1 \right)^2 \right] + \frac{1}{9} \left( \sigma_1 + \sigma_2 + \sigma_3 \right)^2 - p_c^2 = 0
```
We define the following vectors,
```math
    \Delta \sigma = \left[\sigma_1 - \sigma_2 \; , \; \sigma_2 - \sigma_3 \; , \; \sigma_3 - \sigma_1 \right]^T
```
```math
    \Delta \sigma^{cor} = \left[\sigma_1^{cor} - \sigma_2^{cor} \; , \; \sigma_2^{cor} - \sigma_3^{cor} \; , \; \sigma_3^{cor} - \sigma_1^{cor} \right]^T

> ```

Reviewer 2: Naming these differences "Delta" is confusing for me. In return mapping the Delta sigma is usually a stress increment.

Author: What do you suggest for naming?

where
```math
    \sigma^{cor} = \lambda_{cap} C \frac{\partial G_{cap}}{\partial \sigma}
```

Subsituting the stresses with the mapped stresses, we get the following relations.

```math

>     A = \frac{\Delta \sigma^{cor} \cdot \Delta \sigma^{cor}}{2 X^2} + \frac{1}{9} \left( \sum_{i=1}^3{\sigma_i^{cor}} \right)^2

Reviewer 2: Then \sum sigma_i / 3 looks like p^{cor}, but that would lead to a confusing mix of principal stresses and p,q stress invariants.

Author: \sigma^{cor} is defined beforehand, in line 265. 
However, if you think it is still confusing, what do you suggest?
